model_compression_toolkit/core/common/graph/memory_graph/memory_graph.py

```diff
@@ -0,0 +1,151 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import List
+
+from model_compression_toolkit.core.common import Graph, BaseNode
+from model_compression_toolkit.core.common.graph.edge import EDGE_SOURCE_INDEX
+from model_compression_toolkit.core.common.graph.memory_graph.bipartite_graph import DirectedBipartiteGraph
+from model_compression_toolkit.core.common.graph.memory_graph.memory_element import ActivationMemoryTensor
+
+
+class MemoryGraph(DirectedBipartiteGraph):
+    """
+    A representation of a model's graph and intermediate activation tensors as a bipartite graph, to use for schedule
+    and max cut computations.
+    The graph is composed of two sides (sets) of nodes:
+    Side A - the model's graph nodes (operations).
+    Side B - the activation tensors of a computation of the model.
+    An edge e = (a, b) from side A to B exists in the graph if the tensor b is an output of the operation a.
+    An edge e = (b, a) from side B to A exists in the graph if the tensor b is an input of the operation a.
+    """
+
+    def __init__(self, model_graph: Graph):
+        """
+        Args:
+            model_graph: A graph representation of a model.
+        """
+
+        self.model_graph = model_graph
+
+        nodes = list(model_graph.nodes)
+        memory_tensors = []
+        node_to_tensor = []
+        tensor_to_node = []
+
+        for n in nodes:
+            n_outputs = [n.output_shape] if isinstance(n.output_shape, tuple) else n.output_shape
+            out_edges = model_graph.out_edges(n, sort_by_attr=EDGE_SOURCE_INDEX)
+
+            for i, ot in enumerate(n_outputs):
+                memory_tensor = ActivationMemoryTensor(ot, n.name, i)
+                memory_tensors.append(memory_tensor)
+                # Add memory tensor as current node's output
+                node_to_tensor.append((n, memory_tensor))
+
+                ot_edges = [oe for oe in out_edges if oe.source_index == i]
+                for oe in ot_edges:
+                    # Add current memory tensor as input to current node's successors
+                    tensor_to_node.append((memory_tensor, oe.sink_node))
+
+        super().__init__(name=model_graph.name + "_memory_graph",
+                         a_nodes=nodes,
+                         b_nodes=memory_tensors,
+                         edges_ab=[nt for nt in node_to_tensor],
+                         edges_ba=[tn for tn in tensor_to_node])
```

Comment:
why not just pass node_to_tensor and tensor_to_node?

Reply:
You're right, I don't know why I did that, fixed.